packages/cli/src/commands/virtualdevices/events.ts

```diff
@@ -0,0 +1,182 @@
+	const {stateChanges} = output
+	for (const index in input) {
+		const event = input[index]
+		const isStateChange = parseInt(index) < stateChanges.length ? stateChanges[index] : 'undefined'
+		table.push([event.component, event.capability, event.attribute, event.value, isStateChange])
+	}
+	return table.toString()
+}
+
+interface EventInputOutput {
+	input: DeviceEvent[]
+	output: VirtualDeviceEventsResponse
+}
+
+export default class VirtualDeviceEventsCommand extends APICommand<typeof VirtualDeviceEventsCommand.flags> {
+	static description = 'create events for a virtual device\n' +
+		'The command can be run interactively, in question & answer mode, with command line parameters, ' +
+		'or with input from a file or standard in.'
+
+	static examples = [
+		'$ smartthings virtualdevices:events                                                 # interactive mode',
+		'$ smartthings virtualdevices:events <id> -i data.yml                                # from a YAML or JSON file',
+		'$ smartthings virtualdevices:events <id> switch:switch on                           # command line input',
+		'$ smartthings virtualdevices:events <id> temperatureMeasurement:temperature 22.5 C  # command line input',
+	]
+
+	static flags = {
+		...APICommand.flags,
+		...inputAndOutputItem.flags,
+	}
+
+	static args = [
+		{
+			name: 'id',
+			description: 'the device id',
+		},
+		{
+			name: 'name',
+			description: 'the fully qualified attribute name [<component>]:<capability>:<attribute>',
+		},
+		{
+			name: 'value',
+			description: 'the attribute value',
+		},
+		{
+			name: 'unit',
+			description: 'optional unit of measure',
+		},
+	]
+
+	async run(): Promise<void> {
+		const config = {
+			primaryKeyName: 'deviceId',
+			sortKeyName: 'label',
+			listTableFieldDefinitions: ['label', 'name', 'type', 'deviceId'],
+		}
+
+		const deviceId = await selectFromList(this, config, {
+			preselectedId: this.args.id,
+			listItems: () => this.client.devices.list({type: DeviceIntegrationType.VIRTUAL}),
+		})
+
+		const createEvents = async (_: void, input: DeviceEvent[]): Promise<EventInputOutput> => {
+			const output = await this.client.virtualDevices.createEvents(deviceId, input)
+			return {
+				input,
+				output,
+			}
+		}
+		await inputAndOutputItem<DeviceEvent[], EventInputOutput>(this, {
+			buildTableOutput: (data: EventInputOutput) => buildTableOutput(this.tableGenerator, data),
+		}, createEvents, inputProcessor(() => true, () => this.getInputFromUser(deviceId)))
+	}
+
+	async getInputFromUser(deviceId: string): Promise<DeviceEvent[]> {
+		const attributeName = this.args.name
+		const attributeValue = this.args.value
+		let events: DeviceEvent[] = []
+
+		if (attributeName) {
+			if (attributeValue) {
+				const event = await this.parseDeviceEvent(attributeName, attributeValue, this.args.unit)
+				events = [event]
+			} else {
+				throw new Error('Attribute name specified without attribute value')
```

Comment:
We should use `this.error()` instead here.

Reply:
👍